src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +957,835 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+
+                //Perform an insertion in the linking table if required
+                if (multipleCreateStructure.IsLinkingTableInsertionRequired)
+                {
+                    if (multipleCreateStructure.LinkingTableParams is null)
+                    {
+                        multipleCreateStructure.LinkingTableParams = new Dictionary<string, object?>();
+                    }
+
+                    // Consider the mutation request:
+                    // mutation{
+                    //     createbook(item: {
+                    //         title: "Book Title",
+                    //         publisher_id: 1234,
+                    //         authors: [
+                    //             {...} ,
+                    //             {...}
+                    //         ]
+                    //      }) {
+                    //          ...
+                    //      }
+                    // There exists two relationships for a linking table.
+                    // 1. Relationship between the higher level entity (Book) and the linking table.
+                    // 2. Relationship between the current entity (Author) and the linking table.
+                    // To construct the insert database query for the linking table, relationship fields from both the
+                    // relationships are required. 
+
+                    // Populate Current entity's relationship fields
+                    List<ForeignKeyDefinition> foreignKeyDefinitions = currentEntityRelationshipMetadata!.TargetEntityToFkDefinitionMap[multipleCreateStructure.HigherLevelEntityName];
+                    ForeignKeyDefinition fkDefinition = foreignKeyDefinitions[0];
+                    PopulateReferencingFields(sqlMetadataProvider, multipleCreateStructure, fkDefinition, multipleCreateStructure.CurrentEntityCreatedValues, isLinkingTable: true);
+
+                    string linkingEntityName = GraphQLUtils.GenerateLinkingEntityName(multipleCreateStructure.HigherLevelEntityName, entityName);
+                    SourceDefinition linkingTableSourceDefinition = sqlMetadataProvider.GetSourceDefinition(linkingEntityName);
+
+                    _ = BuildAndExecuteInsertDbQueries(sqlMetadataProvider: sqlMetadataProvider,
+                                                       entityName: linkingEntityName,
+                                                       higherLevelEntityName: entityName,
+                                                       parameters: multipleCreateStructure.LinkingTableParams!,
+                                                       sourceDefinition: linkingTableSourceDefinition,
+                                                       isLinkingEntity: true);
+                }
+
+                // Process referencing relationships
+                foreach (Tuple<string, object?> referencingRelationship in multipleCreateStructure.ReferencingRelationships)
```

Comment:
Helps with readability?
```suggestion
                foreach ((string relationshipFieldName, object? relationshipFieldValue) referencingRelationship in multipleCreateStructure.ReferencingRelationships)
```

Reply:
updated